nvflare/private/fed/server/server_engine.py

```diff
@@ -95,32 +97,38 @@ def __init__(self, server, args, client_manager: ClientManager, snapshot_persist
 
         # self.job_def_manager = job_def_manager  # todo: need to figure out how to initialize job manager with inputs
         self.job_def_manager = SimpleJobDefManager(
-            StudyManagerSpec(), FilesystemStorage(root_dir="/workspace/nvflare_provis")
+            StudyManager(FilesystemStorage()), FilesystemStorage(root_dir="/workspace/nvflare_provis")
         )
 
         self.asked_to_stop = False
         self.snapshot_persistor = snapshot_persistor
-        self.command_conn = None
-        self.child_process = None
+        self.parent_conn = None
+        self.parent_conn_lock = Lock()
 
     def _get_server_app_folder(self):
         return "app_server"
 
     def _get_client_app_folder(self, client_name):
         return "app_" + client_name
 
-    def _get_run_folder(self):
-        return os.path.join(self.args.workspace, "run_" + str(self.run_number))
+    def _get_run_folder(self, run_destination):
```

Comment:
These funcs should be promoted into a common util so they can be used in other places. See my previous comments - these lines of code are repeated many times.

Reply:
done